src/gui/addnewtorrentdialog.cpp

```diff
@@ -186,30 +185,20 @@ void AddNewTorrentDialog::setTopLevel(bool value)
 
 int AddNewTorrentDialog::savePathHistoryLength()
 {
-    return savePathHistoryLengthSetting();
+    const int defaultHistoryLength = 8;
+    return settings()->loadValue(KEY_SAVEPATHHISTORYLENGTH, defaultHistoryLength).toInt();
```

Comment:
We also should check the value for min/max before return it.

Reply:
Fixed